src/i_video.c

```diff
@@ -1814,16 +1817,9 @@ static void CreateSurfaces(int w, int h)
     }
 
     // [FG] create intermediate ARGB frame buffer
-    {
-        uint32_t rmask, gmask, bmask, amask;
-        int bpp;
 
-        SDL_PixelFormatEnumToMasks(SDL_PIXELFORMAT_ARGB8888, &bpp,
-                                   &rmask, &gmask, &bmask, &amask);
-        argbbuffer =
-            SDL_CreateRGBSurface(0, w, h, bpp, rmask, gmask, bmask, amask);
-        SDL_FillRect(argbbuffer, NULL, 0);
-    }
+    argbbuffer = SDL_CreateRGBSurfaceWithFormatFrom(
```

Comment:
Instead of `SDL_CreateRGBSurfaceWithFormatFrom(NULL, ...)` you may just want to use `SDL_CreateRGBSurfaceWithFormat()`.

Reply:
The idea is not to allocate `pixels` for `argbbuffer`, so we need to use `SDL_CreateRGBSurfaceWithFormatFrom(NULL, ...)`.

Reply:
Ah, sure, thanks!